[5.7] use new ivar/macro symbol kinds

QuietMisdreavus opened the pull request:
This is a cherry-pick of https://github.com/apple/swift-docc/pull/152 onto `release/5.7`.

**Explanation**: A change in SymbolKit (https://github.com/apple/swift-docc-symbolkit/pull/31) is adding new symbol kinds: ivar and macro. This change updates DocC to handle these new symbols and curate them within rendered documentation.

**Scope**: Adds new behavior to load and curate new symbol types. Existing documentation should render with no change in behavior.

**Radar**: rdar://92124246

**Risk**: Low. This is a small enhancement without changing existing behavior for loading and curating symbols.

**Testing**: A new automated test has been added to ensure that the new symbol kinds can be loaded and organized into the documentation hierarchy correctly.

QuietMisdreavus commented:
@swift-ci Please est

QuietMisdreavus commented:
@swift-ci Please test

ethan-kusters commented on Package.swift:
```diff
@@ -124,7 +124,7 @@ if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
         .package(name: "CLMDB", url: "https://github.com/apple/swift-lmdb.git", .branch("main")),
         .package(url: "https://github.com/apple/swift-argument-parser", .upToNextMinor(from: "1.0.1")),
-        .package(name: "SymbolKit", url: "https://github.com/apple/swift-docc-symbolkit", .branch("main")),
+        .package(name: "SymbolKit", url: "https://github.com/apple/swift-docc-symbolkit", .branch("revert-25-revert-23-5.7/objc-kinds")),
```
When we update this to point at `release/5.7` can we also update `CLMDB` and `swift-markdown` to point at `release/5.7`? This shouldn't change the commit hash since `main` and `release/5.7` point to the same commit but we should still be pointing to the correct branch here.

QuietMisdreavus replied:
Yeah! I can do that once i land the SymbolKit PR.

QuietMisdreavus commented:
I've updated the PR to point back at SymbolKit's `release/5.7` branch, now that https://github.com/apple/swift-docc-symbolkit/pull/31 has landed. I've also updated Package.swift to point to CLMDB's and swift-markdown's `release/5.7` branches as well. Swift-Markdown has had some commits in both `main` and `release/5.7` that haven't been reflected in an update here, so i made sure to keep the same commit in the Package.resolved file. That update can happen in a separate PR.

@swift-ci Please test

QuietMisdreavus commented:
@swift-ci Please test macOS
